src/EditorFeatures/CSharpTest/ConvertToInterpolatedString/ConvertConcatenationToInterpolatedStringTests.cs

```diff
@@ -927,6 +927,48 @@ class C
         const string World = ""World"";
         Console.WriteLine($""{Hello} {World}"");
     }
+}");
+        }
+
+        [WorkItem(42195, "https://github.com/dotnet/roslyn/issues/42195")]
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsConvertToInterpolatedString)]
+        public async Task TestConcatenAtionWithExistingInterpolation()
+        {
+            await TestInRegularAndScriptAsync(
+@"public class C
+{
+    string M(string a, string b)
+    {
+        return [|$""Foo {a} goo: $ "" + b|];
+    }
+}",
+@"public class C
+{
+    string M(string a, string b)
+    {
+        return $""Foo {a} goo: $ {b}"";
+    }
+}");
+        }
+
+        [WorkItem(42195, "https://github.com/dotnet/roslyn/issues/42195")]
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsConvertToInterpolatedString)]
+        public async Task TestConcatenAtionWithExistingInterpolation1()
+        {
+            await TestInRegularAndScriptAsync(
+@"public class C
+{
+    string M(string a, string b, string c)
+    {
+        return [|$""Foo {a} goo: ""+ b + $"" {c} : test""|];
```

Comment:
can we have a couple of tests added where the different string literal parts may or may not be `@` strings?